### PR TITLE
[API Extractor] (Sub-branch) Groundwork for member trimming

### PR DESCRIPTION
**Pull request by pgonzal**
This is a PR into the branch for PR #560.  I'm separating this into two reviews, because we don't want to merge into master until the entire feature is stable.

### CODE REVIEW

**Comment by iclanton** [comment]
>     const resolvedFileOrFolderPath: string = path.resolve(fileOrFolderPath);

Consider calling fs.realPathSync here.

---
Refers to: libraries/node-core-library/src/PackageJsonLookup.ts:79 in e9a528d. [](commit_id = e9a528dcf508c16c02cfd5e85df443a76f86644b, deletion_comment = False)

**Comment by iclanton** [comment]
>     const resolvedFileOrFolderPath: string = path.resolve(fileOrFolderPath);

Or maybe not.

---
In reply to: [373880680](https://github.com/Microsoft/web-build-tools/pull/566#issuecomment-373880680) [](ancestors = 373880680)

---
Refers to: libraries/node-core-library/src/PackageJsonLookup.ts:79 in e9a528d. [](commit_id = e9a528dcf508c16c02cfd5e85df443a76f86644b, deletion_comment = False)

File: apps/api-extractor/src/generators/packageTypings/AstSymbolTable.ts [hunk 1]
```diff
@@ -53,6 +73,19 @@ export class AstSymbolTable {
         throw new Error('Unable to find a root declaration for ' + sourceFile.fileName);
       }
 
+      if (!this._packageMetadataManager.isAedocSupportedFor(sourceFile.fileName)) {
+        const packageJsonPath: string | undefined = this._packageJsonLookup
+          .tryGetPackageJsonFilePathFor(sourceFile.fileName);
+
+        if (packageJsonPath) {
+          throw new Error(`Please add a field such as "tsdoc": { "tsdocFlavor": "AEDoc" } to this file:\n`
```
Comment (iclanton) [inline apps/api-extractor/src/generators/packageTypings/AstSymbolTable.ts]: >Error [](start = 20, length = 5)

Should this only happen in "strict" mode?

**Comment by iclanton** [comment]
Approved.

**Comment by pgonzal** [comment]
>     const resolvedFileOrFolderPath: string = path.resolve(fileOrFolderPath);

I found a way to handle this while still allowing nonexistent paths:  As we're walking upwards, as soon as we encounter a path that DOES exist, we call fsx.realpathSync() exactly once.  Similarly we now call path.resolve() exactly once.

---
In reply to: [373880850](https://github.com/Microsoft/web-build-tools/pull/566#issuecomment-373880850) [](ancestors = 373880850,373880680)

---
Refers to: libraries/node-core-library/src/PackageJsonLookup.ts:79 in e9a528d. [](commit_id = e9a528dcf508c16c02cfd5e85df443a76f86644b, deletion_comment = False)

**Pull request merged**